src/gpio.rs

```diff
@@ -293,1040 +377,841 @@ macro_rules! gpio {
+    fn _set_low(&mut self) {
+        // NOTE(unsafe) atomic write to a stateless register
+        unsafe { (*Gpio::<P>::ptr()).bsrr.write(|w| w.bits(1 << (16 + N))) }
+    }
 
-                    /// Configures the pin to operate as an open-drain output pin.
-                    /// Initial state will be low.
-                    #[inline]
-                    pub fn into_open_drain_output(
-                        self,
-                        cr: &mut $CR,
-                    ) -> $PXi<Output<OpenDrain>> {
-                        self.into_open_drain_output_with_state(cr, State::Low)
-                    }
+    #[inline(always)]
+    fn _is_set_low(&self) -> bool {
+        unsafe { (*Gpio::<P>::ptr()).odr.read().bits() & (1 << N) == 0 }
+    }
+    #[inline(always)]
+    fn _is_low(&self) -> bool {
+        // NOTE(unsafe) atomic read with no side effects
+        unsafe { (*Gpio::<P>::ptr()).idr.read().bits() & (1 << N) == 0 }
+    }
+}
 
-                    /// Configures the pin to operate as an open-drain output pin.
-                    /// `initial_state` specifies whether the pin should be initially high or low.
-                    #[inline]
-                    pub fn into_open_drain_output_with_state(
-                        mut self,
-                        cr: &mut $CR,
-                        initial_state: State,
-                    ) -> $PXi<Output<OpenDrain>> {
-                        self.set_state(initial_state);
-                        unsafe {
-                            $PXi::<Output<OpenDrain>>::set_mode(cr)
-                        }
-                    }
-                    /// Configures the pin to operate as an push-pull output pin.
-                    /// Initial state will be low.
-                    #[inline]
-                    pub fn into_push_pull_output(
-                        self,
-                        cr: &mut $CR,
-                    ) -> $PXi<Output<PushPull>> {
-                        self.into_push_pull_output_with_state(cr, State::Low)
-                    }
+impl<MODE, CR, const P: char, const N: u8> Pin<MODE, CR, P, N>
+where
+    MODE: Active,
+{
+    /// Erases the pin number from the type
+    #[inline]
+    pub fn erase_number(self) -> PEPin<MODE, P> {
+        PEPin::new(N)
+    }
+}
 
-                    /// Configures the pin to operate as an push-pull output pin.
-                    /// `initial_state` specifies whether the pin should be initially high or low.
-                    #[inline]
-                    pub fn into_push_pull_output_with_state(
-                        mut self,
-                        cr: &mut $CR,
-                        initial_state: State,
-                    ) -> $PXi<Output<PushPull>> {
-                        self.set_state(initial_state);
-                        unsafe {
-                            $PXi::<Output<PushPull>>::set_mode(cr)
-                        }
-                    }
+// embedded_hal impls
```

Comment:
This comment looks wrong, we are not `impl`ing `embedded_hal` in this block, right?